assignment2/assignment-2/src/model/questionModel.js

```diff
@@ -0,0 +1,96 @@
+import {EventEmitter} from "events";
+import RestClient from "../rest/RestClient";
+import userModel from "./userModel";
+
+
+ class QuestionModel extends EventEmitter {
+     constructor() {
+         super();
+         this.state = {
+             questions: [{
+                title: "titlu frumos",
+                text: "Cum fac asta?",
+                author: "alexpopa09",
+                tags: "cool tags are for cool kids",
+                creationDate: Date("2019-09-16")
+             },{
+                 title: "lalala",
+                 text: "Ajutor",
+                 author: "maria97",
+                 tags:"tags cool ceva",
+                 creationDate: Date("2018-10-2")
+             
+             }],
+             newQuestion: {
+                title: "",
+                text: "",
+                tags: ""
+             }
+            };
+         }
+
+         loadQuestions(){               
+                const client = new RestClient(userModel.state.newUser.Username, userModel.state.newUser.Password);
+                return client.loadAllQuestions().then(questions => {
+                    this.state = {
+                        ...this.state,
+                        questions: questions
+                    };
+                    this.emit("change", this.state);
+                })
+         }
+
+         addQuestion(title, text, author, tags){
+            const client = new RestClient(userModel.state.newUser.Username, userModel.state.newUser.Password);
+            const question = {
+                id:0,
+                title: title,
+                text: text,
+                author: author,
+                tags: tags
+            } ;
+            console.log(JSON.stringify(question));
+            return client.addQuestion(question).then(() => {this.emit("change",this.state);});
+            
+         }
+
+         filterTitle(title){
+            const client = new RestClient(userModel.state.newUser.Username, userModel.state.newUser.Password);
+            return client.filterTitle(title).then(questions => {
+                this.state = {
+                    ...this.state,
+                    questions: questions
+                };
+                this.emit("change", this.state);
+            })
+            
+         }
+         filterTag(tag){
+            const client = new RestClient(userModel.state.newUser.Username, userModel.state.newUser.Password);
```

Comment:
Would be nicer if you either keep a reference to the client (such that you don't create a new one each time), or maybe move the creation in a different method so you don't have this code all over the place :)